util/cost/estimator.go

```diff
@@ -0,0 +1,73 @@
+package cost
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+var costDenominators = map[v1.ResourceName]int64{
```

Comment:
would it be possible to track `nvidia.com/gpu` resource as well?  I realize not everyone is running gpu device plugins, but many ML jobs leverage gpu resources, which often are the main driver for cost since they are so expensive.

Reply:
sure - could you provide an examples of the Pod YAML please?

Reply:
Yep, here's the -oyaml for a running pod in a 1.16.4 cluster:
```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    sidecar.istio.io/inject: "false"
  creationTimestamp: "2020-01-30T15:54:20Z"
  name: cuda-container
  namespace: ddseapy
  resourceVersion: "68850843"
  selfLink: /api/v1/namespaces/ddseapy/pods/cuda-container
  uid: 4e515cdc-4902-4617-a697-4c64ee15e650
spec:
  containers:
  - args:
    - "100000"
    command:
    - sleep
    image: ubuntu
    imagePullPolicy: Always
    name: cuda-container
    resources:
      limits:
        nvidia.com/gpu: "1"
      requests:
        nvidia.com/gpu: "1"
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-85f6l
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: gpu05
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  - effect: NoSchedule
    key: nvidia.com/gpu
    operator: Exists
  volumes:
  - name: default-token-85f6l
    secret:
      defaultMode: 420
      secretName: default-token-85f6l
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2020-01-30T15:54:20Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2020-01-30T15:54:24Z"
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2020-01-30T15:54:24Z"
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2020-01-30T15:54:20Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://8465a7f8d572fe49b2409d48bd78a3bd42558861c21cd7445c448e12e8525ba4
    image: ubuntu:latest
    imageID: docker-pullable://ubuntu@sha256:8d31dad0c58f552e890d68bbfb735588b6b820a46e459672d96e585871acc110
    lastState: {}
    name: cuda-container
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2020-01-30T15:54:23Z"
  hostIP: 534.125.1.22
  phase: Running
  podIP: 10.33.0.2
  podIPs:
  - ip: 10.33.0.2
  qosClass: BestEffort
  startTime: "2020-01-30T15:54:20Z"
```

Reply:
I expected it to be `gpu` not `nvidia.com/gpu`. Do you think different GPU would be different?

Reply:
yeah, `amd` would be different: https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/#using-device-plugins

but almost everyone uses nvidia for ML because of cuda

Reply:
oh, you mean in terms of pricing.... yeah probably.  The same can be true for cpu though, as they are not all the same.

Reply:
i think the main reason for the difference is not pricing, but features.  amd requires opencl, while nvidia can use cuda.  gpu applications need to be written with one in mind.